subsys/bluetooth/host/l2cap.c

```diff
@@ -1475,7 +1475,11 @@ static void le_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 
 	LOG_DBG("psm 0x%02x scid 0x%04x mtu %u mps %u credits %u", psm, scid, mtu, mps, credits);
 
-	if (mtu < L2CAP_LE_MIN_MTU || mps < L2CAP_LE_MIN_MPS) {
+	/* Validate parameters. Requirements are from (Core Spec v6.0, Vol 3.A.4.22). Valid credit
+	 * range is from 0 to UINT16_MAX.
+	 */
+	if (!IN_RANGE(mtu, L2CAP_LE_MIN_MTU, UINT16_MAX) ||
```

Comment:
The comment mentions a valid credit range (using 0 as the minimal value), but then implements a MTU range check. 

Ideally we'd also define and use a `BT_L2CAP_MAX_MTU` instead of `UINT16_MAX` for the MTU range check.

Reply:
The "initial credit" field of the request L2CAP_LE_CREDIT_BASED_CONNECTION_REQ (Core 4.22) and L2CAP_CREDIT_BASED_CONNECTION_REQ (Core 4.25) has respectively the range 0->UINT16_MAX and 1->UINT16_MAX, which is... confusing. 
I added the comment to this parameter check as we don't need to check it as the entire uint16 range is valid (but a reader might wonder why we don't check it). For the case below, the requirements are taken from 4.25 (as mentioned in the comment), where the credit range starts at 1.

Could perhaps clarify instead: "Valid credit range is from 0 to UINT16_MAX, thus the credit parameter does not need validation"?